QA: fix inconsistency

jrfnl opened the pull request:
## Summary

This PR can be summarized in the following changelog entry:
_N/A_

## Relevant technical choices:

The function documentation states that the return value should be an int, so `null` should be `0` in that case.

Alternatively the function docblock could be adjusted, but to be fair, this seemed like the more logical choice.

As it is now, the unit tests for the `WPSEO_Link_Columns::column_content()` will fail because of this change, so I need some clarification over the desired behaviour to either amend the `WPSEO_Link_Columns::column_content()` method or to amend the unit test.


## Test instructions

_N/A_

moorscode commented:
The way this works is as intended, as there is a difference between 0 and `null` in representation in the table.

I would suggest changing the return value to `int|null` to cover the case that the count is irrelevant instead of `0` returned items.

jrfnl commented:
@moorscode Thanks for the feedback. I've adjusted the documentation now instead (amended the original commit).
I've also added a second commit which deals with the method calling this function actually handling the return value correctly and making sure that if there is output, the output is properly "escaped".
